crates/sui-json-rpc-types/src/sui_event.rs

```diff
@@ -260,33 +257,18 @@ pub enum EventFilter {
         #[serde_as(as = "BigInt<u64>")]
         end_time: u64,
     },
-
-    All(Vec<EventFilter>),
-    Any(Vec<EventFilter>),
-    And(Box<EventFilter>, Box<EventFilter>),
-    Or(Box<EventFilter>, Box<EventFilter>),
 }
 
-impl EventFilter {
-    fn try_matches(&self, item: &SuiEvent) -> SuiResult<bool> {
-        Ok(match self {
+impl Filter<SuiEvent> for EventFilter {
```

Comment:
This PR is removing `Any`, `Not`, `Package`, `EventField` which were not supported by either Fullnode or PG JSON-RPC, but they are part of the (deprecated) events subscription system on fullnodes. What is our policy on features like that?

Reply:
We are already well past the release where we said we'd initially remove support for subscriptions, they are effectively deprecated so we can probably choose to either remove these compound filters from the code and we can highlight that the feature is deprecated and support will continue to be pruned down till its removed, or we just leave it as is. In either case I think we should make sure that these filters are completely removed from documentation, i've found that they've been a constant source of confusion in the docs since they generally aren't supported